### DISCUSSION TITLE
Code/script for appending name of subfolder

### DISCUSSION THREAD

**Discussion by OofMon2301** [SilentVoid13/Templater#1013] [Help] [answered]
I am not quite sure how for Templater to display a subfolder without displaying the parent folder.

For example, ```<% tp.file.folder(true) %>``` displays ```FolderA/FolderB```. Is there a way to only display ```FolderB```?

Apologies if the solution is simple; not good at programming in any case but forcing myself to learn slowly. :>

**Comment by welpdx** [comment, marked as answer]
For this type of thing, I like to use the current file's tFile and then get the parent folder's name:
```js
this.app.workspace.getActiveFile().parent.name
```

Alternatively, you can just manipulate the text from `tp.file.folder()` and get the last substring after the backslash:
```js
tp.file.folder(true).split('/').pop()
```



<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/E7PbcFB.gif" alt="demo" />
</details>


~welp

**Reply by OofMon2301** [reply]
Thank you for the help! Although I'm not sure how to use ``` this.app.workspace.getActiveFile().parent.name ``` or even where to use it.

**Reply by welpdx** [reply]
@OofMon2301 you can use it the same way you use templater functions. You can see how I used it in the demo above. (Just click on `demo` to view gif.

ex:
```js
<% this.app.workspace.getActiveFile().parent.name %>
```

btw, the reason why I prefer this is because it is from the Obsidian API, the same API that templater uses. 


For example, you can see here that Templater's `tp.config.active_file()` is basically just `app.workspace.getActiveFile()` which is from the official [Obsidian API](https://github.com/obsidianmd/obsidian-api/blob/38dd22168d2925086371bfc59e36fd9121527a39/obsidian.d.ts#L4146):

https://github.com/SilentVoid13/Templater/blob/487805b5ad1fd7fbc145040ed82b4c41fc2c48e2/src/core/Templater.ts#L62


this function returns a TFile that gives you information about the file. Having the TFile object of the note you are working with gives you alot of information and options. 


gl!

~welp

**Comment by OofMon2301** [comment]
Sorry to bother again, but I've encountered a few problems that I don't know how to solve.

### **Proposed Plan No. 1**

I want Templater to output the name of the folder that the note is linked from. 

For example, I want the note _Theories of Thermodynamics_ that is in _2023/ENGG1100_ link to a note that is in _2023/Definitions_, but output the name of the folder as **_ENGG1100_**.

![image](https://user-images.githubusercontent.com/114561394/216618638-568bb7b2-92fa-490c-bd71-096145c6d431.png)


This command allows to return the last substring after the backslash:
```javascript
tp.file.folder(true).split('/').pop()
```
But using this command will output the folder _Definitions_.

### *Issue No. 2*

When using other community plugins (such as _Commander_ and _QuickAdd_), some functions stop working as intended. 

For example, function for setting the cursor to a location will not work and will output the code:
```javascript
 <% tp.file.cursor(1) %>
 ``` 

https://user-images.githubusercontent.com/114561394/216621940-dcc2f4d9-27e7-40a3-b0a2-41bdbe5f327b.mp4

Not sure how to fix it or what to do :<.

**Reply by welpdx** [reply]
Hey @OofMon2301,
Please make two separate posts so I can respond to each individually. It is kind of hard to do that here. Please also include the answers to below.

Issue 1: Proposed Plan No. 1
Issue 2: Issue No. 2


Questions:
For Issue 1
1. Do you have some kind of folder template that runs on the creation of a new note? 
2. Are you creating the new note "1st law of thermo" by clicking on the internal link in "Theories of Thermo"?

For Issue 2
1. What is that new lecture button? I dont have that in my vault o.o
